packages/sdk/src/extensions/integrations/instances/balancerV2Liquidity.ts

```diff
@@ -0,0 +1,284 @@
+  poolId,
+  bptAmount,
+  usedTokens,
+  usedTokenAmounts,
+  request,
+  stakingToken,
+}: BalancerV2LiquidityLendAndStakeArgsAndUnstakeAndRedeemArgs): Hex {
+  return encodeAbiParameters(balancerV2LiquidityLendAndStakeEncodingAndUnstakeEncodingAndUnstakeAndRedeemEncoding, [
+    stakingToken,
+    poolId,
+    bptAmount,
+    usedTokens,
+    usedTokenAmounts,
+    request,
+  ]);
+}
+
+export function decodeBalancerV2LiquidityLendAndStakeArgsAndUnstakeAndRedeemArgs(
+  callArgs: Hex,
+): BalancerV2LiquidityLendAndStakeArgsAndUnstakeAndRedeemArgs {
+  const [stakingToken, poolId, bptAmount, usedTokens, usedTokenAmounts, request] = decodeAbiParameters(
+    balancerV2LiquidityLendAndStakeEncodingAndUnstakeEncodingAndUnstakeAndRedeemEncoding,
+    callArgs,
+  );
+
+  return {
+    stakingToken,
+    poolId,
+    bptAmount,
+    usedTokens: [...usedTokens],
+    usedTokenAmounts: [...usedTokenAmounts],
+    request: { ...request, assets: [...request.assets], limits: [...request.limits] },
+  };
+}
+
+export const balancerV2LiquidityClaimRewardsEncoding = [
+  {
+    name: "stakingToken",
+    type: "address",
+  },
+] as const;
+
+export type BalancerV2LiquidityClaimRewardsArgs = {
+  stakingToken: Address;
+};
+
+export function encodeBalancerV2LiquidityClaimRewardsArgs({ stakingToken }: BalancerV2LiquidityClaimRewardsArgs): Hex {
+  return encodeAbiParameters(balancerV2LiquidityClaimRewardsEncoding, [stakingToken]);
+}
+
+export function decodeBalancerV2LiquidityClaimRewardsArgs(integrationData: Hex): BalancerV2LiquidityClaimRewardsArgs {
+  const [stakingToken] = decodeAbiParameters(balancerV2LiquidityClaimRewardsEncoding, integrationData);
+
+  return { stakingToken };
+}
+
+export const balancerV2LiquidityStakeEncodingAndUnstakeEncoding = [
+  {
+    name: "stakingToken",
+    type: "address",
+  },
+  {
+    name: "bptAmount",
+    type: "uint256",
+  },
+] as const;
+
+export type BalancerV2LiquidityStakeArgsAndUnstakeArgs = {
+  stakingToken: Address;
+  bptAmount: bigint;
+};
+
+export function encodeBalancerV2LiquidityStakeArgsAndUnstakeArgs({
+  stakingToken,
+  bptAmount,
+}: BalancerV2LiquidityStakeArgsAndUnstakeArgs): Hex {
+  return encodeAbiParameters(balancerV2LiquidityStakeEncodingAndUnstakeEncoding, [stakingToken, bptAmount]);
+}
+
+export function decodeBalancerV2LiquidityStakeArgsAndUnstakeArgs(
+  integrationData: Hex,
+): BalancerV2LiquidityStakeArgsAndUnstakeArgs {
+  const [stakingToken, bptAmount] = decodeAbiParameters(
+    balancerV2LiquidityStakeEncodingAndUnstakeEncoding,
+    integrationData,
+  );
+
+  return { stakingToken, bptAmount };
+}
+
+export const SwapKind = {
+  GivenIn: 0,
+  GivenOut: 1,
+} as const;
+
+export type SwapKindValue = typeof SwapKind[keyof typeof SwapKind];
+
+export function assertSwapKindValue(value: number): asserts value is SwapKindValue {
+  if (!Object.values(SwapKind).includes(value as SwapKindValue)) {
+    throw new Error(`Invalid SwapKind: ${value}`);
+  }
+}
```

Comment:
1. This should be internal (not exported and maybe not even a standalone function)
2. Is there no better way to deal with this case?